src/mdns.rs

```diff
@@ -0,0 +1,444 @@
+use esp_idf_sys::*;
+
+extern crate alloc;
+use alloc::vec::Vec;
+
+use std::ffi::CStr;
+use std::ffi::CString;
+
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use std::time::Duration;
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Interface {
+    STA,
+    AP,
+    ETH,
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Protocol {
+    V4,
+    V6,
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Type {
+    A = MDNS_TYPE_A as _,
+    AAAA = MDNS_TYPE_AAAA as _,
+    ANY = MDNS_TYPE_ANY as _,
+    NSEC = MDNS_TYPE_NSEC as _,
+    OPT = MDNS_TYPE_OPT as _,
+    PTR = MDNS_TYPE_PTR as _,
+    SRV = MDNS_TYPE_SRV as _,
+    TXT = MDNS_TYPE_TXT as _,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct QueryResult {
```

Comment:
Please use the String type from the `alloc` module, not its std:: reexport. Otherwise this code will not compile under no_std. Or better yet: use `heapless::String` if these strings have upper bounds in terms of max length
 This way you won't depend for them on neither `alloc` nor `std` which would allow to move these types or a generalized version of those to `embedded-svc`, where no-std and no-alloc is a requirement.

Reply:
We used alloc module.